plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -370,22 +470,66 @@ export class MongoDBInstrumentation extends InstrumentationBase<
   /**
    * Populate span's attributes by fetching related metadata from the context
    * @param span span to add attributes to
+   * @param connectionCtx mongodb internal connection context
    * @param ns mongodb namespace
    * @param command mongodb internal representation of a command
+   */
+  private _populateV4Attributes(
+    span: Span,
+    connectionCtx: any,
+    ns: mongodb.MongoDBNamespace,
+    command?: any
+  ) {
+    let host, port: undefined | string;
+    if (connectionCtx) {
+      const hostParts =
+        typeof connectionCtx.address === 'string'
+          ? connectionCtx.address.split(':')
+          : '';
+      if (hostParts.length === 2) {
+        host = hostParts[0];
+        port = hostParts[1];
+      }
+    }
+    // capture parameters within the query as well if enhancedDatabaseReporting is enabled.
+    let commandObj: Record<string, unknown>;
+    if (command?.documents) {
+      commandObj = command.documents[0];
```

Comment:
what is the meaning of this documents in this place? can it be empty array?

Reply:
Fixed! ty